src/daos_vol_file.c

```diff
@@ -2014,10 +2016,6 @@ H5_daos_duns_resolve_path_comp_cb(tse_task_t *task, void H5VL_DAOS_UNUSED *args)
         } /* end if */
     } /* end if */
     else {
-        /* Verify that this is actually an HDF5 DAOS file */
-        if(udata->duns_attr.da_type != DAOS_PROP_CO_LAYOUT_HDF5)
-            D_GOTO_ERROR(H5E_FILE, H5E_NOTHDF5, -H5_DAOS_BAD_VALUE, "file '%s' is not a valid HDF5 DAOS file", udata->path);
```

Comment:
It seems like this change would prevent the connector from being able to detect and give an error for when someone tries to open a native file. Is there another way that this should be done through the DUNS?

Reply:
the issue is that the DUNS can be used with 2 modes:
- like we do now which is fine
- using a URL style: daos://pooluuid/contuuid/
the issue is the second way there is no stub type, and this check needs to be done through the cont prop which i plan to add in a second patch.
I guess my question is why this check is required if the other operations like fetching the root group or metadata object would fail if this is not an HDF5 file?

Reply:
if you prefer to have some sort of check here, we can see if the pool uuid returned here is valid, and that could give us some check that this is not a native file.

Reply:
> I guess my question is why this check is required if the other operations like fetching the root group or metadata object would fail if this is not an HDF5 file?

The check isn't required, it's more of just a nice thing for the connector user. If the root group/metadata object fetch failed, that might lead someone to believe something is wrong with their file, whereas this check just gives a bit more information on what's actually wrong.

> if you prefer to have some sort of check here, we can see if the pool uuid returned here is valid, and that could give us some check that this is not a native file.

If that change lets us keep a similar check while still making this work with the URL-style DUNS path, that would be preferable to me at least since I think the error message is helpful. On the other hand, proper compatibility with the DUNS is probably more important.

Reply:
ok will update this to check for proper pool uuid format instead.
anyway, i think if this is a native file, we wouldn't even be here i would think and the resolve path would have returned an error.
i will add a proper way to check for the container type in a different PR that utilizes the actual layout property that is set when the container is created.

Reply:
> ok will update this to check for proper pool uuid format instead.

Thanks!

> anyway, i think if this is a native file, we wouldn't even be here i would think and the resolve path would have returned an error.

I've definitely hit one of these checks before when testing specifically what happens when you try to open a native file. It may not have been this check in particular (I believe this is replicated elsewhere), but I can at least verify we can get to these checks.